include/boost/geometry/strategy/cartesian/side_by_triangle.hpp

```diff
@@ -205,23 +206,27 @@ public :
     template <typename P1, typename P2, typename P>
     static inline int apply(P1 const& p1, P2 const& p2, P const& p)
     {
-        using coor_t = typename select_calculation_type_alt<CalculationType, P1, P2, P>::type;
-
-        // Promote float->double, small int->int
-        using promoted_t = typename select_most_precise<coor_t, double>::type;
-
-        bool const are_all_integral_coordinates =
+        constexpr bool are_all_integral_coordinates =
             std::is_integral<coordinate_type_t<P1>>::value
             && std::is_integral<coordinate_type_t<P2>>::value
             && std::is_integral<coordinate_type_t<P>>::value;
 
+        // Promote float->double, integers to std::int64_t
+        using coor_t = typename select_calculation_type_alt<CalculationType, P1, P2, P>::type;
+        using promoted_t = std::conditional_t
+            <
+                are_all_integral_coordinates,
+                typename promote_integral<coor_t>::type,
+                typename select_most_precise<coor_t, double>::type
+            >;
+
         eps_policy< math::detail::equals_factor_policy<promoted_t> > epsp;
-        promoted_t s = compute_side_value
+        promoted_t const s = compute_side_value
             <
                 coor_t, promoted_t, are_all_integral_coordinates
             >::apply(p1, p2, p, epsp);
 
-        promoted_t const zero = promoted_t();
+        static promoted_t const zero = promoted_t();
```

Comment:
Is this a microoptimisation with a meaningful benefit or is there some other intention? For primitive types (almost every case, I'd expect), I wouldn't expect it. I suppose this is for custom number types were the initialisation of 0 may be costly.

I find this makes the code less clear, it made me wonder if that value was intended to be referenced/pointed to somewhere outside the scope or something like this, maybe it is just unfamiliarity on my part, though.

Reply:
It's more harmonizing with other code than a micro optimization.
For example, in `segment_ratio.hpp`: `static thistype result(0, 1);`
In `get_turn_info.hpp`: `static decltype(dm.measure) const zero = 0;`
In `infinite_lien_functions.hpp`: `static Type const zero = 0;`

But there are also cases where it is not static.
I intended to make it `constexpr` instead but that is not supported (IIRC) by Boost.MultiPrecision.